lib/LaTeXML/Package/listings.sty.ltxml

```diff
@@ -166,9 +164,15 @@ sub lstProcessBlock {
     [@{ LookupValue('LISTINGS_POSTAMBLE') },
       T_END]); }
 
+our %lstFilenameRPL = (
+  '_' => T_CS('\textunderscore'), '$' => T_CS('\textdollar'), '-' => T_CS('\textendash'));
+
 sub lstProcessDisplay {
-  my ($name, $text) = @_;
+  my ($name, $text)    = @_;
   my ($body, $trailer) = lstProcessBlock($name, $text);
+  # reencode certain chars, so typewriter-like, but not typewriter font.
+  $name = Tokens(map { $lstFilenameRPL{ ToString($_) } || $_; } $name->unlist) if $name;
+  AssignValue('LST@toctitle', $name);    # so it shows up in list of..
```

Comment:
Let's see if I can break this one: does
```latex
\lstset{caption=\lstname}
```
behave as intended? It seems to be like `\lstname` is still incorrect. (Example from https://texdoc.org/serve/listings/0.)

Reply:
And it should be done before `lstProcessBlock` to fix `@dataname` as well.

Reply:
What did you try, and what did you expect? It seems to work for me.

Reply:
> What did you try for `\lstname` and what did you expect? It seems to work for me.

Oh, `\lstname` is set some time in `lstProcessDisplay`, not in `lstActivate`... so it does work. I assume nobody else sets `\lstname` and/or uses `\LST@name`?